### PR TITLE
Replace std::string_view of c++17 with c++11 compatible string-view-lite

### PR DESCRIPTION
**Pull request by clover2123**

### CODE REVIEW

**Comment by clover2123** [comment]
Performance result: there is no impact of this patch

|    test    | walrus_patch [s] | origin_walrus [s]|
|------------|--------------|----------------------------------------|
|   change   |2.598 (1.000x)|             2.583 (0.994x)             |
|  factorial |2.708 (1.000x)|             2.752 (1.016x)             |
|  fannkuch  |2.254 (1.000x)|             2.314 (1.026x)             |
|  fibonacci |3.805 (1.000x)|             3.793 (0.997x)             |
|   gregory  |3.014 (1.000x)|             3.083 (1.023x)             |
|    hanoi   |3.400 (1.000x)|             3.382 (0.995x)             |
|  heapsort  |2.492 (1.000x)|             2.565 (1.029x)             |
|   huffman|3.520 (1.000x)|             3.530 (1.003x)             |
|k_nucleotide|2.934 (1.000x)|             3.006 (1.025x)             |
| mandelbrot |3.866 (1.000x)|             3.967 (1.026x)             |
|    nbody   |3.290 (1.000x)|             3.294 (1.001x)             |
|   nqueens  |3.101 (1.000x)|             3.111 (1.003x)             |
|    prime   |3.338 (1.000x)|             3.368 (1.009x)             |
| quick_sort |2.692 (1.000x)|             2.659 (0.988x)             |
|  red-black |3.614 (1.000x)|             3.685 (1.020x)             |
|  salesman  |3.125 (1.000x)|             3.117 (0.997x)             |

**Comment by clover2123** [comment]
Get c++11 `string-view-list` from https://github.com/martinmoene/string-view-lite with Boost Software License 1.0

**Comment by clover2123** [comment]
License check done and add BSL-1.0 license file

File: build/target.cmake [hunk 1]
```diff
@@ -25,7 +25,7 @@ ENDIF()
 
 # Default options per compiler
 IF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" OR ${COMPILER_CLANG_CL})
-    SET (WALRUS_CXXFLAGS /std:c++17 /fp:strict /Zc:__cplusplus /EHs /source-charset:utf-8 /D_CRT_SECURE_NO_WARNINGS /D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING /wd4244 /wd4267 /wd4805 /wd4018 /wd4172)
+    SET (WALRUS_CXXFLAGS /std:c++11 /fp:strict /Zc:__cplusplus /EHs /source-charset:utf-8 /D_CRT_SECURE_NO_WARNINGS /D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING /wd4244 /wd4267 /wd4805 /wd4018 /wd4172)
```
Comment (ksh8281) [inline build/target.cmake:28]: There is no /std:c++11 on msvc :(
could you change this to /std:c++14?

Comment (clover2123) [reply build/target.cmake]: Fixed it, thanks!

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**